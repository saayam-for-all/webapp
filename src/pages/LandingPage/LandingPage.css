.landing-title {
  font-size: 100px;
  color: rgb(171, 171, 171);
  opacity: 0.8;
  font-weight: 700;
  margin-bottom: -20px;
}
.landing-subtitle {
  color: rgb(198, 197, 197);
  width: fit-content;
  font-size: 20px;
  margin-bottom: 50px;
}
.video-section {
  background-color: #fff; /* Match the background color of other sections */
  overflow: hidden;
  max-width: 100%;
  height: calc(100vh - 100px);
}

.video-wrapper {
  width: 300%;
  left: -100%;
  pointer-events: none;
  position: relative;
  padding-bottom: 85%; /* 16:9 aspect ratio (9 รท 16 = 0.5625 or 56.25%) */
}

.video-wrapper iframe {
  position: absolute;
  top: -60px;
  left: 0;
  width: 100%;
  height: 75%;
  border-radius: 10px; /* Optional: match the border-radius of other sections */
  pointer-events: none;
}

.video-overlay {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 5;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  pointer-events: auto;
}
.landing-volume-button {
  position: absolute;
  bottom: 5%;
  right: 0%;
  z-index: 10;
  color: black;
  padding: 0.5rem 1rem;
  border: none;
  cursor: pointer;
}
.community-container {
  display: flex;
  justify-content: center;
  align-items: flex-start;
  padding: 4rem 3rem;
  gap: 4rem;
  flex-wrap: wrap;
  font-family: "Arial", sans-serif;
}

.cards-section {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  gap: 0.2rem; /* cards are close together */
}

.community-card {
  width: 200px;
  height: 300px;
  border-radius: 25px;
  background-size: cover;
  background-position: center;
  overflow: hidden;
  box-shadow: 6px 6px 12px rgba(0, 0, 0, 0.1);
  border: 2px solid #fff;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  padding-top: 1rem;
}

.card-overlay {
  color: white;
  font-weight: bold;
  font-size: 1rem;
  text-align: center;
  border-radius: 15px;
  padding: 0.5rem 0.8rem;
  width: 85%;
}

.text-section {
  max-width: 400px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  margin-top: 30px; /* aligns with middle card */
  text-align: left;
}

.text-section h2 {
  font-size: 2.5rem;
  font-weight: 700;
  margin-bottom: 1rem;
}

.text-section p {
  font-size: 1.1rem;
  color: #555;
  margin-bottom: 1.5rem;
  line-height: 1.6;
}

.join-btn {
  background-color: #4a67f8;
  color: white;
  border: none;
  padding: 0.8rem 1.5rem;
  border-radius: 25px;
  font-weight: 600;
  font-size: 1rem;
  cursor: pointer;
  transition: background-color 0.3s ease;
  align-self: flex-start;
}

.join-btn:hover {
  background-color: #3c54d1;
}

/* .video-overlay button {
  padding: 0.75rem 1.5rem;
  font-size: 1.2rem;
  background: #ffffffcc;
  border: none;
  border-radius: 8px;
  cursor: pointer;
  transition: background 0.3s ease;
}

.video-overlay button:hover {
  background: #fff;
} */

/* Extra small devices (phones, 600px and down)
@media only screen and (max-width: 600px) {
    .poster_image {
        width: 100%;
    }

    .poster_image img {
        width: 100%;
        height: 300px;

    }

    .poster_image p {
        color: black;
        position: relative;

        font-family: cursive;
    }

    .HomePage_Comp2 .flex-container,
    .HomePage_Comp3 .flex-container {
        padding: 1rem;
    }

    .HomePage_Comp2 .flex-item-right,
    .HomePage_Comp3 .flex-item-left {
        font-family: cursive;
        font-size: 1rem;
    }

    .section_3 {
        font-family: cursive;
        font-size: 1rem;
        text-align: center;
        padding: 2rem;
    }

    .HomePage_Comp1 {
        padding-bottom: 2rem;
    }

    .Section_5_content,
    .Section_3_content {
        width: 90% !important;
        margin-left: 5% !important;
    }
}


@media only screen and (min-width: 600px) {
    .HomePage_Comp1 {
        padding-bottom: 2rem;
    }

    .poster_image {
        width: 100%;
    }

    .poster_image img {
        width: 100%;
        height: 300px
    }

    .poster_image p {
        color: black;
        position: relative;
      
        font-family: cursive;
    }

    .HomePage_Comp2 .flex-container,
    .HomePage_Comp3 .flex-container {
        padding: 1rem;
    }

    .HomePage_Comp2 .flex-item-right,
    .HomePage_Comp3 .flex-item-left {
        font-family: cursive;
        font-size: 1rem;
    }

    .section_3 {
        font-family: cursive;
        font-size: 1rem;
        text-align: center;
        padding: 2rem;
    }

    .Section_5_content,
    .Section_3_content {
        width: 90% !important;
        margin-left: 5% !important;
    }
}


@media only screen and (min-width: 768px) {
    .poster_image {
        width: 75%;
    }

    .poster_image img {
        width: 100%;
        height: 500px
    }

    .poster_image p {
        color: black;
        position: absolute;
       
        width: 75%;
        top: 29rem;
        font-size: 1.5rem;
        font-family: cursive;
    }

    .what_we_do {
        margin-top: -80px !important;
    }

    .what_we_do p {
        font-family: cursive;
        font-size: 1.5rem;
    }

    .HomePage_Comp2 .flex-container,
    .HomePage_Comp3 .flex-container {
        padding: 2.5rem;
    }

    .HomePage_Comp2 .flex-item-right,
    .HomePage_Comp3 .flex-item-left {
        font-family: cursive;
        font-size: 1.2rem;
    }

    .section_3 {
        font-family: cursive;
        font-size: 1rem;
        text-align: center;
        padding: 2rem;
    }

    .Section_5_content,
    .Section_3_content {
        width: 90% !important;
        margin-left: 5% !important;
    }
}


@media only screen and (min-width: 992px) {
    .poster_image {
        width: 75%;
    }

    .poster_image img {
        width: 100%;
        height: 500px
    }

    
    .poster_image p {
        color: black;
        position: absolute;
       
        top: 25rem;
        font-size: 1.9rem;
        font-family: cursive;
        width: 75%;
    }


    .what_we_do {
        margin-top: -100px !important;
    }

    .what_we_do p {
        font-family: cursive;
        font-size: 1.5rem;
    }

    .HomePage_Comp2 .flex-container,
    .HomePage_Comp3 .flex-container {
        border-radius: 50px;
        padding: 3rem;
        box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    }

    .HomePage_Comp2 .flex-item-right,
    .HomePage_Comp3 .flex-item-left {
        font-family: cursive;
        font-size: 1.2rem;
    }

    .section_3 {
        font-family: cursive;
        font-size: 1.4rem;
        text-align: center;
        padding: 3rem;
    }
}


@media only screen and (min-width: 1200px) {

    .poster_image {
        width: 75%;
    }

    .poster_image img {
        width: 100%;
        height: 500px
    }

    .poster_image p {
        color: black;
        position: absolute;
       
        top: 25rem;
        font-size: 2rem;
        font-family: cursive;
        width: 75%;
    }

    .what_we_do {
        margin-top: -50px !important;
    }

    .what_we_do p {
        font-family: cursive;
        font-size: 1.5rem;
    }

    .HomePage_Comp2 .flex-container,
    .HomePage_Comp3 .flex-container {
        padding: 3rem;
        border-radius: 50px;
        box-shadow: 5px 8px 8px 2px rgba(11, 255, 210, 0.2), 0 6px 20px 0 rgba(11, 218, 255, 0.19);
    }

    .HomePage_Comp2 .flex-item-right,
    .HomePage_Comp3 .flex-item-left {
        font-family: cursive;
        font-size: 1.6rem;
    }

    .section_3 {
        font-family: cursive;
        font-size: 1.4rem;
        text-align: center;
        padding: 3rem;
    }
}

.HomePage_Comp2 .flex-container,
.HomePage_Comp3 .flex-container {
    display: flex;
    flex-direction: row;
    font-size: 30px;
    text-align: center;

}

.HomePage_Comp2 .flex-container {
    background-image: linear-gradient(#7670ed, #69ffb6);
}

.HomePage_Comp3 .flex-container {
    background-image: linear-gradient(#ff5d5d, #ffff69);

}

.HomePage_Comp2 .flex-item-left,
.HomePage_Comp3 .flex-item-left {
    flex: 50%;
}

.HomePage_Comp2 .flex-item-right,
.HomePage_Comp3 .flex-item-right {
    flex: 50%;
    align-self: center;
}


@media (max-width: 800px) {

    .HomePage_Comp2 .flex-container,
    .HomePage_Comp3 .flex-container {
        flex-direction: column;
    }
}

.HomePage_Comp2 .flex-item-left img,
.HomePage_Comp3 .flex-item-right img {
    width: 100%;
}

.HomePage_Comp2 .flex-item-right,
.HomePage_Comp3 .flex-item-right {
    padding: 1rem
}

.Section_5_content p {
    font-family: cursive;
    font-size: 1.4rem;
    text-align: center;
    padding: 3rem;
}

.Section_5_content,
.Section_3_content {
    border-radius: 25px;

    margin-top: 4rem;
    margin-left: 25%;
    align-self: center;
    width: 50%;
    color: white;
    background-image: linear-gradient(180deg, #2d6de1, #2d6de1, #2d6de1, #0e4675)
}

.Section_5_content a {
    
    font-family: cursive;
}

.HomePage_Comp4 {
    width: 80%;
    margin-left: 10%;
    margin-top: 2rem;
}

.HomePage_Comp4 p {
    font-family: cursive;
    font-size: large;
    text-align: left;
}

.HomePage_Comp4 h1 {
    font-size: 40px;
}

.Section_3_content {
    margin-bottom: 3rem;
} */

.MainPage_content {
  font-family: cursive;
  font-size: larger;
  width: 75%;
  margin-left: 12.5%;
}

.MainPage_content {
  padding-bottom: 1.8rem;
}

.poster_image img {
  width: 40%;
}

@media only screen and (max-width: 820px) {
  .poster_image img {
    width: 100%;
  }
}

/*embla carousel styles*/
.embla {
  overflow: hidden;
  height: auto;
  width: 60%;
  display: block;
  margin-left: auto;
  margin-right: auto;
}
.embla__container {
  display: flex;
}
.embla__slide {
  flex: 0 0 100%;
  min-width: 0;
}
