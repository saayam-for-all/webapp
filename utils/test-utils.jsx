import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import authReducer from "#redux/features/authentication/authSlice";

// redux state for logged in user
export const MOCK_STATE_LOGGED_IN = {
  auth: { user: { userId: "mockUser" }, idToken: "mockToken" },
};
// redux state for logged out user
export const MOCK_STATE_LOGGED_OUT = { auth: { user: null } };

export function renderWithProviders(ui, extendedRenderOptions = {}) {
  const {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: {
        auth: authReducer,
      },
      preloadedState,
    }),
    ...renderOptions
  } = extendedRenderOptions;

  const Wrapper = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );

  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
}
