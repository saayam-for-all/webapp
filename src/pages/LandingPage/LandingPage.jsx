import React, { useEffect } from "react";
import { useNavigate } from "react-router";
import { useSelector } from "react-redux";
import HeroSection from "./components/HeroSection";
import Info from "./components/Info";
import Images from "./components/Dynamic_img";

import "./LandingPage.css";

export default function Home() {
  const { user } = useSelector((state) => state.auth);
  const navigate = useNavigate();

  useEffect(() => {
    if (user !== null) {
      navigate("/dashboard");
    }
  }, [user]);

  return (
    <div>
      <Images />
      <HeroSection />
      <Info />
    </div>
  );
}
