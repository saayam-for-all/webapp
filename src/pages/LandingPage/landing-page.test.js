import { renderWithProviders } from "#utils/test-utils.jsx";
import LandingPage from "./LandingPage";

// Mock internal dependencies
jest.mock("react-router");

// Mock custom components
jest.mock("./components/HeroSection");
jest.mock("./components/Info");
jest.mock("./components/Dynamic_img");

describe("LandingPage", () => {
  it("renders correctly", () => {
    const tree = renderWithProviders(<LandingPage />);
    expect(tree).toMatchSnapshot();
  });
});
