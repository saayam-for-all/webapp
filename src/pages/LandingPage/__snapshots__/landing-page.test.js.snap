// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full"
      >
        <section
          class="video-section"
        >
          <div
            class="video-overlay flex flex-col justify-center"
          >
            <div
              class="flex flex-col items-end justify-end"
            >
              <p
                class="landing-title"
              >
                Sayaam For All
              </p>
              <h6
                class="landing-subtitle"
              >
                Connecting Help to those who need it
              </h6>
            </div>
            <button
              class="text-white bg-blue-500 hover:bg-blue-700 rounded-lg px-8 py-2 text-lg hover:underline"
            >
              Join the Community
            </button>
          </div>
          <button
            class="landing-volume-button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M301.1 34.8C312.6 40 320 51.4 320 64l0 384c0 12.6-7.4 24-18.9 29.2s-25 3.1-34.4-5.3L131.8 352 64 352c-35.3 0-64-28.7-64-64l0-64c0-35.3 28.7-64 64-64l67.8 0L266.7 40.1c9.4-8.4 22.9-10.4 34.4-5.3zM425 167l55 55 55-55c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-55 55 55 55c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-55-55-55 55c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l55-55-55-55c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0z"
              />
            </svg>
          </button>
          <div
            class="video-wrapper"
          >
            <iframe
              allow="autoplay"
              id="landing-iframe"
              src="https://www.youtube.com/embed/9CBLVoSSuwM?autoplay=1&mute=1&loop=1&controls=0&playlist=9CBLVoSSuwM&enablejsapi=1"
              title="YouTube Video"
            />
          </div>
        </section>
        <div
          class="flex flex-row items-center bg-white h-96 mt-5 gap-10"
        >
          <div
            class="flex flex-col justify-start items-start w-3/5 p-10 h-full"
          >
            <h3
              class="font-bold text-3xl h-1/3 w-2/3"
            >
              Creating a World Where Help is Always Within Reach
            </h3>
            <p
              class="h-1/3 text-lg text-gray-500"
            >
              Through our global network of volunteers and donors, we empower communities to support one another, fostering a culture of compassion and solidarity that transcends geographical boundaries.
            </p>
            <button
              class="mt-10 text-sm text-gray-500 hover:text-blue-500 hover:underline"
            >
              How We Operate →
            </button>
          </div>
          <div
            class="flex flex-row w-1/3 h-full gap-5"
          >
            <img
              alt="Help 1"
              class="w-1/2 rounded-lg"
              src="https://platform.theverge.com/wp-content/uploads/sites/2/2025/03/STK093_GOOGLE_E.jpg?quality=90&strip=all&crop=0%2C0%2C100%2C100&w=2400"
            />
            <img
              alt="Help 2"
              class="w-1/2 rounded-lg"
              src="https://platform.theverge.com/wp-content/uploads/sites/2/2025/03/STK093_GOOGLE_E.jpg?quality=90&strip=all&crop=0%2C0%2C100%2C100&w=2400"
            />
          </div>
        </div>
        <mock-carousel />
        <div
          class="community-container"
        >
          <div
            class="cards-section"
          >
            <div
              class="community-card"
              style="background-image: url(https://platform.theverge.com/wp-content/uploads/sites/2/2025/03/STK093_GOOGLE_E.jpg?quality=90&strip=all&crop=0%2C0%2C100%2C100&w=2400); margin-top: 0px;"
            >
              <div
                class="card-overlay"
              >
                Take help from our Volunteers
              </div>
            </div>
            <div
              class="community-card"
              style="background-image: url(https://platform.theverge.com/wp-content/uploads/sites/2/2025/03/STK093_GOOGLE_E.jpg?quality=90&strip=all&crop=0%2C0%2C100%2C100&w=2400); margin-top: 120px;"
            >
              <div
                class="card-overlay"
              >
                Planning and budgeting advice for travel
              </div>
            </div>
            <div
              class="community-card"
              style="background-image: url(https://platform.theverge.com/wp-content/uploads/sites/2/2025/03/STK093_GOOGLE_E.jpg?quality=90&strip=all&crop=0%2C0%2C100%2C100&w=2400); margin-top: 240px;"
            >
              <div
                class="card-overlay"
              >
                A whole new community of charity organisations
              </div>
            </div>
          </div>
          <div
            class="text-section"
          >
            <h2>
              Discover our community
            </h2>
            <p>
              Chat with our community and get in touch with different charity organisations!
            </p>
            <button
              class="join-btn"
            >
              Join the community
            </button>
          </div>
        </div>
      </div>
    </div>
    <script
      src="https://www.youtube.com/iframe_api"
    />
  </body>,
  "container": <div>
    <div
      class="w-full"
    >
      <section
        class="video-section"
      >
        <div
          class="video-overlay flex flex-col justify-center"
        >
          <div
            class="flex flex-col items-end justify-end"
          >
            <p
              class="landing-title"
            >
              Sayaam For All
            </p>
            <h6
              class="landing-subtitle"
            >
              Connecting Help to those who need it
            </h6>
          </div>
          <button
            class="text-white bg-blue-500 hover:bg-blue-700 rounded-lg px-8 py-2 text-lg hover:underline"
          >
            Join the Community
          </button>
        </div>
        <button
          class="landing-volume-button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M301.1 34.8C312.6 40 320 51.4 320 64l0 384c0 12.6-7.4 24-18.9 29.2s-25 3.1-34.4-5.3L131.8 352 64 352c-35.3 0-64-28.7-64-64l0-64c0-35.3 28.7-64 64-64l67.8 0L266.7 40.1c9.4-8.4 22.9-10.4 34.4-5.3zM425 167l55 55 55-55c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-55 55 55 55c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-55-55-55 55c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l55-55-55-55c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0z"
            />
          </svg>
        </button>
        <div
          class="video-wrapper"
        >
          <iframe
            allow="autoplay"
            id="landing-iframe"
            src="https://www.youtube.com/embed/9CBLVoSSuwM?autoplay=1&mute=1&loop=1&controls=0&playlist=9CBLVoSSuwM&enablejsapi=1"
            title="YouTube Video"
          />
        </div>
      </section>
      <div
        class="flex flex-row items-center bg-white h-96 mt-5 gap-10"
      >
        <div
          class="flex flex-col justify-start items-start w-3/5 p-10 h-full"
        >
          <h3
            class="font-bold text-3xl h-1/3 w-2/3"
          >
            Creating a World Where Help is Always Within Reach
          </h3>
          <p
            class="h-1/3 text-lg text-gray-500"
          >
            Through our global network of volunteers and donors, we empower communities to support one another, fostering a culture of compassion and solidarity that transcends geographical boundaries.
          </p>
          <button
            class="mt-10 text-sm text-gray-500 hover:text-blue-500 hover:underline"
          >
            How We Operate →
          </button>
        </div>
        <div
          class="flex flex-row w-1/3 h-full gap-5"
        >
          <img
            alt="Help 1"
            class="w-1/2 rounded-lg"
            src="https://platform.theverge.com/wp-content/uploads/sites/2/2025/03/STK093_GOOGLE_E.jpg?quality=90&strip=all&crop=0%2C0%2C100%2C100&w=2400"
          />
          <img
            alt="Help 2"
            class="w-1/2 rounded-lg"
            src="https://platform.theverge.com/wp-content/uploads/sites/2/2025/03/STK093_GOOGLE_E.jpg?quality=90&strip=all&crop=0%2C0%2C100%2C100&w=2400"
          />
        </div>
      </div>
      <mock-carousel />
      <div
        class="community-container"
      >
        <div
          class="cards-section"
        >
          <div
            class="community-card"
            style="background-image: url(https://platform.theverge.com/wp-content/uploads/sites/2/2025/03/STK093_GOOGLE_E.jpg?quality=90&strip=all&crop=0%2C0%2C100%2C100&w=2400); margin-top: 0px;"
          >
            <div
              class="card-overlay"
            >
              Take help from our Volunteers
            </div>
          </div>
          <div
            class="community-card"
            style="background-image: url(https://platform.theverge.com/wp-content/uploads/sites/2/2025/03/STK093_GOOGLE_E.jpg?quality=90&strip=all&crop=0%2C0%2C100%2C100&w=2400); margin-top: 120px;"
          >
            <div
              class="card-overlay"
            >
              Planning and budgeting advice for travel
            </div>
          </div>
          <div
            class="community-card"
            style="background-image: url(https://platform.theverge.com/wp-content/uploads/sites/2/2025/03/STK093_GOOGLE_E.jpg?quality=90&strip=all&crop=0%2C0%2C100%2C100&w=2400); margin-top: 240px;"
          >
            <div
              class="card-overlay"
            >
              A whole new community of charity organisations
            </div>
          </div>
        </div>
        <div
          class="text-section"
        >
          <h2>
            Discover our community
          </h2>
          <p>
            Chat with our community and get in touch with different charity organisations!
          </p>
          <button
            class="join-btn"
          >
            Join the community
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
